ocaml/xapi/helpers.ml

```diff
@@ -1016,6 +1016,16 @@ let get_all_plugged_srs ~__context =
   let pbds_plugged_in = List.filter (fun self -> Db.PBD.get_currently_attached ~__context ~self) pbds in
   List.setify (List.map (fun self -> Db.PBD.get_SR ~__context ~self) pbds_plugged_in)
 
+let get_all_plugged_srs_local ~__context =
+  let localhost = get_localhost __context in
+  let localhost = Ref.string_of localhost in
+  let my_pbds_plugged_in = Db.PBD.get_refs_where  ~__context ~expr:(And (
```

Comment:
You could use `get_all_where` also in the function above, to reduce the Db calls also on the rest of `get_all_plugged_srs` invocations

Reply:
This sounds like a good idea :) Is there a reason you suggested `get_all_where` here while I used `get_refs_where` in the new function - and the `List.setify` line following is the same? Is `get_all_where` prefered when you're getting a lot of records?

Reply:
You are right, you just need `get_refs_where`

Reply:
It may be worth checking if getting all the relevant records is faster than making all the separate call to get the SRs or not.

Reply:
I'm happy to take this PR as it is already an improvement. I have only a minor comment: To my ear `get_all_local_plugged_srs` is easier to understand than `get_all_plugged_srs_local` or would that distort the meaning in some way? If not, I would like to suggest this change.

Reply:
How about `get_local_plugged_srs`? I think that's the most clear, agreed the local at the end is a bit awkard.

Reply:
Sounds good to me.